fido/fido.py

```diff
@@ -56,6 +92,20 @@ def dataReceived(self, data):
         self.buffer.write(data)
 
     def connectionLost(self, reason):
+        """
+        :param reason:
```

Comment:
This might be out of scope/not useful, but is there a handy web link to the Twisted docs for why we are checking for these specific reasons? Are these the only reasons to check? Are we not checking some reasons?

This is dense code and pointing to more docs could be helpful.